src/main/java/net/imagej/ops/deconvolve/RichardsonLucyC.java

```diff
@@ -0,0 +1,184 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imagej.ops.deconvolve;
+
+import net.imagej.ops.Ops;
+import net.imagej.ops.filter.AbstractIterativeFFTFilterC;
+import net.imagej.ops.special.computer.AbstractUnaryComputerOp;
+import net.imagej.ops.special.computer.BinaryComputerOp;
+import net.imagej.ops.special.computer.Computers;
+import net.imglib2.Cursor;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.img.Img;
+import net.imglib2.outofbounds.OutOfBoundsConstantValueFactory;
+import net.imglib2.outofbounds.OutOfBoundsFactory;
+import net.imglib2.type.Type;
+import net.imglib2.type.numeric.ComplexType;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.util.Util;
+import net.imglib2.view.Views;
+
+import org.scijava.Priority;
+import org.scijava.app.StatusService;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+/**
+ * Richardson Lucy algorithm for (@link RandomAccessibleInterval) (Lucy, L. B.
+ * (1974).
+ * "An iterative technique for the rectification of observed distributions".)
+ * 
+ * @author Brian Northan
+ * @param <I>
+ * @param <O>
+ * @param <K>
+ * @param <C>
+ */
+
+@Plugin(type = Ops.Deconvolve.RichardsonLucy.class,
+	priority = Priority.HIGH_PRIORITY)
+public class RichardsonLucyC<I extends RealType<I>, O extends RealType<O>, K extends RealType<K>, C extends ComplexType<C>>
+	extends AbstractIterativeFFTFilterC<I, O, K, C> implements
+	Ops.Deconvolve.RichardsonLucy
+{
+
+	@Parameter(required = false)
+	private StatusService status;
+
+	/**
+	 * An OutOfBoundsFactory which defines the extension strategy
+	 */
+	@Parameter(required = false)
+	private OutOfBoundsFactory<O, RandomAccessibleInterval<O>> obfOutput;
+
+	/**
+	 * Op that computes Richardson Lucy update
+	 */
+	@Parameter(required = false)
+	private AbstractUnaryComputerOp<RandomAccessibleInterval<O>, RandomAccessibleInterval<O>> update =
+		null;
+
+	BinaryComputerOp<RandomAccessibleInterval<I>, RandomAccessibleInterval<O>, RandomAccessibleInterval<O>> rlCorrection;
+
+	@Override
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public void initialize() {
+		super.initialize();
+
+		if (update == null) {
+			update = (AbstractUnaryComputerOp) Computers.unary(ops(),
+				RichardsonLucyUpdate.class, RandomAccessibleInterval.class,
+				RandomAccessibleInterval.class);
+		}
+
+		rlCorrection = (BinaryComputerOp) Computers.binary(ops(),
+			RichardsonLucyCorrection.class, RandomAccessibleInterval.class,
+			RandomAccessibleInterval.class, RandomAccessibleInterval.class,
+			getFFTInput(), getFFTKernel());
+
+	}
+
+	@Override
+	public void performIterations(RandomAccessibleInterval<I> in,
+		RandomAccessibleInterval<K> kernel, RandomAccessibleInterval<O> out)
+	{
+
+		createReblurred();
```

Comment:
maybe return reblurred `RAI` for later usage? Assumptions over the state of a `super-class` are hard to understand for others.